src/Domain/ElementDistribution.hpp

```diff
@@ -0,0 +1,111 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <utility>
+#include <vector>
+
+#include "Domain/Structure/ElementId.hpp"
+
+namespace domain {
+
+/*!
+ * \brief Distribution strategy for assigning elements to CPUs using a
+ * Morton('Z-order') space-filling curve to determine placement within each
+ * block.
+ *
+ * \details The element distribution is computed by first performing a greedy
+ * assignment of a balanced number of elements to each processor by giving each
+ * processor an allowance of [total number of elements]/[number of processors]
+ * elements from one or more blocks, and is given from blocks starting with the
+ * lowest number block that still has elements to contribute to an allowance.
```

Comment:
This sentence is veeeeery long, and I think splitting it would improve readability. Here's a suggested split, but please feel free to do a different one:

The element distribution assigns a balanced number of elements to each processor. This distribution is computed by first greedily assigning to each processor an allowance of [total number of elements]/[number of processors] elements from one or more blocks, starting with the lowest number block that still has elements to contribute to an allowance.